### PR TITLE
refactor: update grenade.rs to use bones timer.

### LINKED ISSUE #739: Update grenade to use Bones time resource instead of manually setting the time
We can change the `grenade` object to use the bones time resource instead of manually setting the time. Similar to `musket` and `invisibility`.

### PR DESCRIPTION
**Pull request by lvllvl**
updated grenade.rs to use bones timer instead of manually setting time to 4.0 seconds

attempts to resolve #739

### CODE REVIEW

File: core/src/elements/grenade.rs [hunk 1]
```diff
@@ -138,7 +141,12 @@ fn update_idle_grenades(
             commands.add(
                 move |mut idle: CompMut<IdleGrenade>, mut lit: CompMut<LitGrenade>| {
                     idle.remove(entity);
-                    lit.insert(entity, LitGrenade { age: 0.0 });
+                    lit.insert(
+                        entity,
+                        LitGrenade {
+                            damage_delay: Timer::new(Duration::from_secs_f32(4.0), TimerMode::Once),
```
Comment (zicklag) [inline core/src/elements/grenade.rs:147]: This `4.0` should come from the `fuse_time` out of the grenade meta.

File: core/src/elements/grenade.rs [hunk 2]
```diff
@@ -215,7 +224,7 @@ fn update_lit_grenades(
         }
 
         // If it's time to explode
-        if grenade.age >= *fuse_time {
+        if grenade.damage_delay.elapsed_secs() >= *fuse_time {
```
Comment (zicklag) [inline core/src/elements/grenade.rs:227]: Once we set the timer duratino from the `fuse_time`, we can just do this:

```suggestion
        if grenade.damage_delay.is_finished() {
```

**Review by zicklag — CHANGES_REQUESTED** [review]

File: core/src/elements/grenade.rs [hunk 3]
```diff
@@ -12,11 +13,12 @@ pub fn install(session: &mut GameSession) {
...
 pub struct IdleGrenade;
 
-#[derive(Clone, TypeUlid, Debug, Copy)]
+// #[derive(Clone, TypeUlid, Debug, Copy)]
+#[derive(Clone, TypeUlid, Debug)]
 #[ulid = "01GPY9N9CBR6EFJX0RS2H2K58J"]
 pub struct LitGrenade {
     /// How long the grenade has been lit.
-    pub age: f32,
+    pub damage_delay: Timer,
```
Comment (zicklag) [inline core/src/elements/grenade.rs:21]: Last change! Could you re-name this to `fuse_time`, and remove the comment, or update it with a new description.

**Review by zicklag — CHANGES_REQUESTED** [review]

**Review by zicklag — APPROVED** [review]
Looks good!

**Comment by lvllvl** [comment]
> Looks good!

thanks for the help !

**Pull request merged**